ibis-server/app/model/data_source.py

```diff
@@ -24,31 +26,59 @@ def get_connection(self, dto):
                 raise NotImplementedError(f'Unsupported data source: {self}')
 
     @staticmethod
-    def get_postgres_connection(dto: PostgresDTO):
-        return ibis.postgres.connect(
-            user=dto.user,
-            password=dto.password,
-            database=dto.database,
-            host=dto.host,
-            port=dto.port,
-        )
+    def get_postgres_connection(info: 'PostgresConnectionInfo') -> BaseBackend:
+        if info.jdbc_url:
+            resource = info.jdbc_url.removeprefix("jdbc:")
+        else:
+            resource = f"postgres://{info.user}:{info.password}@{info.host}:{info.port}/{info.database}"
+        return ibis.connect(resource)
 
     @staticmethod
-    def get_bigquery_connection(dto: BigQueryDTO):
-        credits_json = loads(base64.b64decode(dto.credentials).decode('utf-8'))
+    def get_bigquery_connection(info: 'BigQueryConnectionInfo') -> BaseBackend:
+        credits_json = loads(base64.b64decode(info.credentials).decode('utf-8'))
         credentials = service_account.Credentials.from_service_account_info(credits_json)
         return ibis.bigquery.connect(
-            project_id=dto.project_id,
-            dataset_id=dto.dataset_id,
+            project_id=info.project_id,
+            dataset_id=info.dataset_id,
             credentials=credentials,
         )
 
     @staticmethod
-    def get_snowflake_connection(dto: SnowflakeDTO):
+    def get_snowflake_connection(info: 'SnowflakeConnectionInfo') -> BaseBackend:
         return ibis.snowflake.connect(
-            user=dto.user,
-            password=dto.password,
-            account=dto.account,
-            database=dto.database,
-            schema=dto.sf_schema,
+            user=info.user,
+            password=info.password,
+            account=info.account,
+            database=info.database,
+            schema=info.sf_schema,
         )
+
+
+class PostgresConnectionInfo(BaseModel):
+    host: Optional[str] = Field(examples=["localhost"], default=None)
+    port: Optional[int] = Field(default=5432)
+    database: Optional[str] = None
+    user: Optional[str] = None
+    password: Optional[str] = None
+    jdbc_url: Optional[str] = None
```

Comment:
```suggestion
    connection_string: Optional[str] = None
```
It's not a jdbc url actually.